crates/cdk-lnbits/src/lib.rs

```diff
@@ -352,12 +355,27 @@ impl MintPayment for LNbits {
                 Self::Err::Anyhow(anyhow!("Could not check invoice status"))
             })?;
 
-        Ok(vec![WaitPaymentResponse {
-            payment_identifier: payment_identifier.clone(),
-            payment_amount: Amount::from(payment.details.amount as u64),
-            unit: CurrencyUnit::Sat,
-            payment_id: payment.details.payment_hash,
-        }])
+        let amount = payment.details.amount;
+
+        if amount == i64::MIN {
+            return Err(Error::AmountOverflow.into());
+        }
+
+        match payment.paid {
+            true => Ok(vec![WaitPaymentResponse {
+                payment_identifier: payment_identifier.clone(),
+                payment_amount: Amount::from(amount.unsigned_abs()),
+                unit: CurrencyUnit::Msat,
+                payment_id: payment.details.payment_hash,
+            }]),
+
+            false => Ok(vec![WaitPaymentResponse {
+                payment_identifier: payment_identifier.clone(),
+                payment_amount: Amount::ZERO,
+                unit: CurrencyUnit::Msat,
+                payment_id: payment.details.payment_hash,
+            }]),
```

Comment:
Should we just return an empty list here?

Reply:
Yeah looking at cln unpaid are filtered out so think its best to do the same here.